api/v1alpha1/resourcegraphdefinition_types.go

```diff
@@ -16,7 +16,24 @@ package v1alpha1
 import (
 	extv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/runtime"
+	runtime "k8s.io/apimachinery/pkg/runtime"
+)
+
+// AdditionalPrinterColumnPolicy defines how additional printer columns are applied to the generated CustomResourceDefinition (CRD). This policy controls whether user-provided columns replace or merge with the default columns.
+type AdditionalPrinterColumnPolicy string
+
+const (
+	// AdditionalPrinterColumnPolicyReplace specifies that the provided additional
+	// printer columns should completely replace the default columns. If no
+	// AdditionalPrinterColumns are provided or the list is empty, the system
+	// will fall back to using the default columns.
+	AdditionalPrinterColumnPolicyReplace AdditionalPrinterColumnPolicy = "Replace"
```

Comment:
I don't think we should expose this. IMO there should be only supported way that behaves similarly to CRDs:
- If a user defines an `additionalPrinterColumns`, it gets added to the columns
- If a user overrides a column, there should be an additional column with the same title (like how CRD works with `Name`)
- There should probably be a policy forbidding to override KRO's defined columns (STATE, READY, AGE). That's an addition to CRDs, but it doesn't change the behaviour.

I see no reason where the KRO's added printed columns should be removed.
Kro objects should have consistent column data.

A side note that the proposed `Add` is confusing. It is currently some kind of `Merge` where the definition of a column content can be overridden rather than `Add`